src/utils/js-yaml-loader.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+const fs = require( 'fs' );
+const path = require( 'path' );
+const yaml = require( 'js-yaml' );
+const defaultOptions = require( '../default-options' );
+const utils = require( './utils' );
+const C = require( '../constants/constants' );
+const argv = require( 'minimist' )( process.argv.slice( 2 ) );
+const LOG_LEVEL_KEYS = Object.keys( C.LOG_LEVEL );
+
+function readAndParseFile( filePath, callback ) {
+	try {
+		fs.readFile( filePath, 'utf8', function( error, fileContent ) {
+			if ( error ) {
+				return callback ( error );
+			}
+			try {
+				const config = parseFile( filePath, fileContent );
+				return callback( null, config );
+
+			} catch ( error ) {
+				return callback ( error );
+			}
+		} );
+	} catch ( error ) {
+		return callback ( error );
+	}
+}
+
+function parseFile( filePath, fileContent ) {
+	if ( fileContent == null ) {
+		fileContent = fs.readFileSync( filePath, {encoding: 'utf8'} );
```

Comment:
Would be better to have a seperate readFileSync function and have parseFile just parse the file